unix/read_stubs.c

```diff
@@ -0,0 +1,70 @@
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/custom.h>
+#include <caml/callback.h>
+#include <caml/alloc.h>
+#include <caml/unixsupport.h>
+#include <caml/bigarray.h>
+#include <caml/threads.h>
+
+#include <stdio.h>
+#include <errno.h>
+
+CAMLprim value stub_cstruct_read(value val_fd, value val_c)
```

Comment:
Can this be marked as noalloc ? I think unix_error allocs !?

Reply:
definitely can't be noalloc -- it releases the runtime lock as well.